docs/pubsub.md

```diff
@@ -38,271 +38,289 @@ Scope app (01.app.yaml) :
-       log.Fatal(http.ListenAndServe(addr, s.Mux))
-   }
+func (s *Server) Run(addr string) { // this is called by the main()
+    fmt.Printf("pubsub api is up! Listening on port: %s\n", addr)
+    log.Fatal(http.ListenAndServe(addr, s.Mux))
+}
 ```
 
-Init method (api/controller/base.go) : This is responsible for allocating the server muxer and inserting our handler on it.
+Init method (api/controller/base.go) : This is responsible for allocating the server multiplexer (mux) and inserting our handler on it.
 
 ```go
    // Init - configures the server
-   func (s *Server) Init() {
-       s.Mux = http.NewServeMux()
-       client := dappclient.NewAppClient()
-       s.Mux.HandleFunc("/publish", func(w http.ResponseWriter, r *http.Request) {
-           ctx, cancel := context.WithCancel(context.Background())
-           defer cancel()
-           data := Message{}
-           decoder := json.NewDecoder(r.Body)
-
-           err := decoder.Decode(&data)
-           if err != nil {
-               rest.ERROR(w, err)
-               return
-           }
-
-           discordMsg := models.Message{
-               Data: data.Message,
-           }
-           if err := client.WriteMessage(ctx, discordCH, discordMsg); err != nil {
-               fmt.Println(err)
-               rest.ERROR(w, err)
-           }
-
-           rest.JSON(w, http.StatusOK, nil)
-       })
-   }
+func (s *Server) Init() {
+	s.Mux = http.NewServeMux()
+	client := dappclient.NewAppClient()
+	s.Mux.HandleFunc("/publish", func(w http.ResponseWriter, r *http.Request) {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		data := message{}
+		decoder := json.NewDecoder(r.Body)
+
+		err := decoder.Decode(&data)
+		if err != nil {
+			rest.ERROR(w, err)
+			return
+		}
+
+		pubMsg := data.Message
+		if err := client.WriteMessage(ctx, pubsubChannel, pubMsg); err != nil {
+			fmt.Println(err)
+			rest.ERROR(w, err)
+		}
+
+		rest.JSON(w, http.StatusOK, "Message sent!")
+	})
+}
 ```
 
 ## Subscribers
 
-As most applications that run on Inspr the subscribers only execute their tasks when they receive a message through it's dApp client. Because of this, their structure is characteristic, an endless loop with a read from some desired Channel. These programs must also have access to the structure the client is expecting and be able to convert the simple text message it will receive to this format. Once the message has been sent to the clients the message is committed, acknowledging to the message broker that that application is done with that message and can go on to the next one.
+As most applications that run on Inspr the subscribers only execute their tasks when they receive a message through it's dApp client. Once the message has been sent to the clients the message is committed, acknowledging to the message broker that that application is done with that message and can go on to the next one.
```

Comment:
remove the extra "that" in "acknowledging to the message broker that that application"